Error in GitLab actions causes  `panic`

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa5e3d3]

goroutine 1 [running]:
github.com/updatecli/updatecli/pkg/plugins/resources/gitlab/mergerequest.(*Gitlab).CreateAction(0xc000e129a0, {0xc000e705c0, 0x3d}, {0xc000a20a00?, 0x25c7f40?}, {0xc000286900?, 0x1?})
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/gitlab/mergerequest/create.go:82 +0x473
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).RunActions(0xc000235e40)
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/actions.go:150 +0xa16
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).Run(0xc000235e40)
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/main.go:233 +0x3ce
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).Run(0x491c080)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/main.go:269 +0x234
github.com/updatecli/updatecli/cmd.run({0x2b50743?, 0x0?})
        /home/runner/work/updatecli/updatecli/cmd/root.go:91 +0x3c6
github.com/updatecli/updatecli/cmd.glob..func1(0x48d2520?, {0x2b4db02?, 0x1?, 0x1?})
        /home/runner/work/updatecli/updatecli/cmd/apply.go:29 +0x10e
github.com/spf13/cobra.(*Command).execute(0x48d2520, {0xc000705320, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x48d16c0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/updatecli/updatecli/cmd.Execute()
        /home/runner/work/updatecli/updatecli/cmd/root.go:43 +0x65
main.main()
        /home/runner/work/updatecli/updatecli/main.go:8 +0x17
```

### Expected Behavior

Graceful failure

I think the failure is the because the [change_log](https://github.com/open-policy-agent/opa/releases/tag/v0.50.0) returned is too big so gitlab API rejects it.

Because reordering these blocks [create.go#L82-L92](https://github.com/updatecli/updatecli/blob/main/pkg/plugins/resources/gitlab/mergerequest/create.go#L82-L92) (which is probably the fix)

Reports this:

```text
Pipeline "Open Policy Agent" failed
Skipping due to:
	action stage:	"Post \"https:/<longurl....>\": http2: server sent GOAWAY and closed the connection; LastStreamID=37, ErrCode=ENHANCE_YOUR_CALM, debug=\"\""
```


### Steps To Reproduce

Using the following source (who's changelog caused the issue):

```
name: Open Policy Agent
sources:
  latestVersion:
    name: Get latest opa release
    kind: githubrelease
    spec:
      owner: open-policy-agent
      repository: opa
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      versionFilter:
        kind: semver
    transformers:
      - trimprefix: v
```

### Environment

```markdown
- OS: WSL (Ubuntu 20.04)
- updatecli: 0.47.0
```


### Anything else?

_No response_

--- a/pkg/plugins/resources/gitlab/mergerequest/create.go
+++ b/pkg/plugins/resources/gitlab/mergerequest/create.go
@@ -79,16 +79,16 @@ func (g *Gitlab) CreateAction(title, changelog, pipelineReport string) error {
 		&opts,
 	)
 
-	if resp.Status > 400 {
-		logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
-	}
-
 	if err != nil {
 		if err.Error() == scm.ErrNotFound.Error() {
 			logrus.Infof("Gitlab pullrequest not created, skipping")
 			return nil
 		}
 		return err
+	}
+
+	if resp.Status > 400 {
+		logrus.Debugf("RC: %d\nBody:\n%s", resp.Status, resp.Body)
 	}
 
 	logrus.Infof("Gitlab pullrequest successfully opened on %q", pr.Link)

